sys/shell/commands/sc_netif.c

```diff
@@ -25,227 +27,343 @@
+            printf("source address length");
+            break;
+
+        case NETCONF_OPT_CHANNEL:
+            printf("channel");
+            break;
+
+        case NETCONF_OPT_NID:
+            printf("network identifier");
+            break;
+
+        default:
+            /* we don't serve these options here */
+            break;
     }
 }
 
-void _netif_send(int argc, char **argv)
+void _netif_list(kernel_pid_t dev)
 {
-    kernel_pid_t dev;
-    uint8_t addr[MAX_ADDR_LEN];
-    uint8_t addr_len;
-    ng_pktsnip_t *pkt;
-    ng_netif_hdr_t *nethdr;
+    uint8_t hwaddr[MAX_ADDR_LEN];
+    uint16_t u16;
+    int res;
 
+    printf("Iface %2d  ", dev);
 
-    if (argc < 4) {
-        printf("usage: %s IF ADDR DATA\n", argv[0]);
-        return;
+    res = ng_netapi_get(dev, NETCONF_OPT_ADDRESS, 0, hwaddr, sizeof(hwaddr));
+
+    if (res >= 0) {
+        printf(" HWaddr: ");
+        _print_hwaddr(hwaddr, (uint8_t)res);
+        printf(" ");
     }
-    /* parse interface */
-    dev = (kernel_pid_t)atoi(argv[1]);
-    if (thread_get(dev) == NULL) {
-        puts("error: invalid interface given");
-        return;
+
+    res = ng_netapi_get(dev, NETCONF_OPT_CHANNEL, 0, &u16, sizeof(u16));
+
+    if (res >= 0) {
+        printf(" Channel: %" PRIu16 " ", u16);
     }
-    /* parse address */
-    addr_len = _parse_addr(argv[2], addr);
-    if (addr_len == 0) {
-        puts("error: invalid address given");
-        return;
+
+    res = ng_netapi_get(dev, NETCONF_OPT_NID, 0, &u16, sizeof(u16));
+
+    if (res >= 0) {
+        printf(" NID: %" PRIx16 " ", u16);
     }
-    /* put packet together */
-    pkt = ng_pktbuf_add(NULL, argv[3], strlen(argv[3]), NG_NETTYPE_UNDEF);
-    pkt = ng_pktbuf_add(pkt, NULL, sizeof(ng_netif_hdr_t) + addr_len,
-                     NG_NETTYPE_UNDEF);
-    nethdr = (ng_netif_hdr_t *)pkt->data;
-    ng_netif_hdr_init(nethdr, 0, addr_len);
-    ng_netif_hdr_set_dst_addr(nethdr, addr, addr_len);
-    /* and send it */
-    ng_netapi_send(dev, pkt);
+
+    printf("\n            ");
+
+    res = ng_netapi_get(dev, NETCONF_OPT_ADDRESS_LONG, 0, hwaddr, sizeof(hwaddr));
+
+    if (res >= 0) {
+        printf("Long HWaddr: ");
+        _print_hwaddr(hwaddr, (uint8_t)res);
+        printf("\n            ");
+    }
+
+    res = ng_netapi_get(dev, NETCONF_OPT_SRC_LEN, 0, &u16, sizeof(u16));
+
+    if (res >= 0) {
+        printf("Source address length: %" PRIx16 "\n            ", u16);
+    }
+
+    /* TODO: list IPv6 info */
+
+    puts("");
 }
 
-void _netif_addr(int argc, char **argv)
+static void _netif_set_u16(kernel_pid_t dev, ng_netconf_opt_t opt,
+                           char *u16_str)
 {
-    kernel_pid_t dev;
-    uint8_t addr[MAX_ADDR_LEN];
-    size_t addr_len;
-    int res;
+    unsigned int res;
+    bool hex = false;
 
-    if (argc < 2) {
-        printf("usage: %s IF [ADDR]\n", argv[0]);
-        return;
+    if (_is_number(u16_str)) {
+        if ((res = strtoul(u16_str, NULL, 10)) == ULONG_MAX) {
```

Comment:
I would prefer to specify addresses always in hex format (e.g. xx:xx -> a3:0f), so for short and long addresses, this way there is no confusion about what format is used...